### PR TITLE
Implement home_mount_path logic

### LINKED ISSUE #2891: Allow unix-tasks to create symlinks in another directory
I'd like to implement an NFS home directory mounting structure where the user's home is automounted at eg `/u` and the
tasks daemon would create the appropriate symlink in `/home`. I think this is currently impossible. Are there security implications for allowing this? If not, could I try my hand at a PR?

### PR DESCRIPTION
**Pull request by antonl**
Fixes #2891 

The clippy integration doesn't seem to pass, but that's unrelated to the changes.

Checklist

- [x] This pr contains no AI generated code
- [x] cargo fmt has been run
- [ ! ] cargo clippy has been run 
- [x] cargo test has been run and passes
- [ ] book chapter included (if relevant)
- [ ] design document included (if relevant)

### CODE REVIEW

File: examples/unixd [hunk 1]
```diff
@@ -60,6 +60,12 @@ pam_allowed_login_groups = ["posix_group"]
 
 # home_attr = "uuid"
 
+# The prefix prepended to where home directories will be mounted to. If unset, defaults to the value
```
Comment (Firstyear) [inline examples/unixd:63]: minor, but could we worth expanding here what this does with an example like your use case you have to help people "see" what it does.

**Review by Firstyear — COMMENTED** [review]
wow, you smashed this out so fast. well done! Please also add yourself to CONTRIBUTORS.md :)

**Comment by antonl** [comment]
I'm getting build errors when building with the selinux feature:
```rust
error[E0432]: unresolved import `crate::selinux_util`
  --> unix_integration/resolver/src/bin/kanidm_unixd_tasks.rs:38:5
   |
38 | use crate::selinux_util;
   |     ^^^^^^^^^^^^^^^^^^^ no `selinux_util` in the root
   |
help: consider importing this module instead
   |
38 | use kanidm_unix_resolver::selinux_util;
   |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

For more information about this error, try `rustc --explain E0432`.
error: could not compile `kanidm_unix_int` (bin "kanidm_unixd_tasks") due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

Any hints in how to troubleshoot this?

**Comment by Firstyear** [comment]
> I'm getting build errors when building with the selinux feature:
> 
> ```rust
> error[E0432]: unresolved import `crate::selinux_util`
>   --> unix_integration/resolver/src/bin/kanidm_unixd_tasks.rs:38:5
>    |
> 38 | use crate::selinux_util;
>    |     ^^^^^^^^^^^^^^^^^^^ no `selinux_util` in the root
>    |
> help: consider importing this module instead
>    |
> 38 | use kanidm_unix_resolver::selinux_util;
>    |     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> For more information about this error, try `rustc --explain E0432`.
> error: could not compile `kanidm_unix_int` (bin "kanidm_unixd_tasks") due to 1 previous error
> warning: build failed, waiting for other jobs to finish...
> ```
> 
> Any hints in how to troubleshoot this? Should I just do what rustc says?

That might be a bug on our side, I think follow the compiler advice and change the use statement.

**Comment by antonl** [comment]
Ok, so it builds after I make the fix but now I get this when trying it live. This could be due to a mismatched server/client version though: v1.2.3 vs develop.
```
ERROR    🚨 [error]: Provider authentication failed | err: Http(500, Some(InvalidAuthState("session id not present in cred presented to 'begin' step")), 
```

File: examples/unixd (hunk 1 above)
Comment (antonl) [reply examples/unixd]: I wrote some stuff, let me know if it makes sense.

**Comment by Firstyear** [comment]
> Ok, so it builds after I make the fix but now I get this when trying it live. This could be due to a mismatched server/client version though: v1.2.3 vs develop.
> 
> ```
> ERROR    🚨 [error]: Provider authentication failed | err: Http(500, Some(InvalidAuthState("session id not present in cred presented to 'begin' step")), 
> ```
> 
> Is there a trick to getting the client/server interaction to run locally?

Yeah, that's a change in a newer version, but it shouldn't have affected this? I'll have to have a test.

**Review by Firstyear — APPROVED** [review]

**Comment by Firstyear** [comment]
@antonl Actually, as a check, we should also be creating the home dir https://github.com/kanidm/kanidm/blob/b3571dc8708509ac4b1f3a7c827d872c736aeaf1/unix_integration/resolver/src/bin/kanidm_unixd_tasks.rs#L89 in home_mount

**Pull request merged**